dev/Expander/Expander.xaml

```diff
@@ -0,0 +1,200 @@
+                                            <ObjectAnimationUsingKeyFrames Storyboard.TargetName="ExpanderContent" Storyboard.TargetProperty="Visibility">
+                                                <DiscreteObjectKeyFrame KeyTime="0" Value="Visible" />
+                                            </ObjectAnimationUsingKeyFrames>
+                                        </Storyboard>
+                                    </VisualState.Storyboard>
+                                </VisualState>
+                                <VisualState x:Name="Collapsed">
+                                    <VisualState.Storyboard>
+                                        <Storyboard>
+                                            <PopOutThemeAnimation TargetName="ExpanderContent" />
+                                        </Storyboard>
+                                    </VisualState.Storyboard>
+                                </VisualState>
+                            </VisualStateGroup>
+                            <VisualStateGroup x:Name="ExpandDirectionStates">
+                                <VisualState x:Name="Down" />
+                                <VisualState x:Name="Up">
+                                    <VisualState.Setters>
+                                        <Setter Target="ExpanderHeader.(Grid.Row)" Value="1" />
+                                        <Setter Target="ExpanderContent.(Grid.Row)" Value="0" />
+                                        <Setter Target="Row0.Height" Value="*" />
+                                        <Setter Target="Row1.Height" Value="Auto" />
+                                        <Setter Target="PopIn.FromVerticalOffset" Value="{ThemeResource ExpanderPopinVerticalOffset}"/>
+                                    </VisualState.Setters>
+                                </VisualState>
+                            </VisualStateGroup>
+                        </VisualStateManager.VisualStateGroups>
+
+                        <Grid.RowDefinitions>
+                            <RowDefinition x:Name="Row0" Height="Auto" />
+                            <RowDefinition x:Name="Row1" Height="*" />
+                        </Grid.RowDefinitions>
+                        <Grid.ColumnDefinitions>
+                            <ColumnDefinition x:Name="Column0" Width="*" />
+                            <ColumnDefinition x:Name="Column1" Width="Auto" />
+                        </Grid.ColumnDefinitions>
+
+                        <ToggleButton
+                            x:Name="ExpanderHeader"
+                            AutomationProperties.AutomationId="ExpanderToggleButton"
+                            MinWidth="0"
+                            MinHeight="0"
+                            IsTabStop="{ThemeResource ExpanderHeaderIsTabStop}"
+                            IsEnabled="{TemplateBinding IsEnabled}"
+                            Style="{StaticResource ExpanderHeaderStyle}"
```

Comment:
We've had a lot of headache in the Toolkit with having this header's style being embedded in the control template itself. I'd suggest to add a `HeaderStyle` property on the control which is then template bound here, with a default in the Style set to this static resource that's defined here. It'll make it easier for a dev to stylize just the header without having to override the template or much with injecting named resources (which is hard to apply across an app).

Reply:
That is good feedback. @ejserna can we take that as a follow up since that involves adding a new API ? Also, I assume the style will have to be updated based on the direction, so do we end up overriding the custom style when the direction changes ?